### PR TITLE
feat: created common package and added new architecture updates.

### PR DESCRIPTION
**Pull request by afsal-ali**
- [x] Create common package
- [x]  Adopt new architecture

https://linear.app/lottiefiles/issue/DOTL-87/create-common-package
https://linear.app/lottiefiles/issue/DOTL-86/adopt-new-architecture

### CODE REVIEW

File: packages/common/src/dotlottie-player.ts [hunk 1]
```diff
@@ -270,7 +307,8 @@ export class DotLottieContainer {
     if (!this._lottie) return;
     this._lottie.addEventListener('enterFrame', () => {
       this.frame.value = this._lottie?.currentFrame ?? 0;
-      this.seeker.value = ((this._lottie?.currentFrame ?? 0) / (this._lottie?.totalFrames ?? 0)) * 100;
+      this.seeker.value =
+        ((this._lottie?.currentFrame ?? 0) / (this._lottie?.totalFrames ?? 0)) * 100;
```
Comment (samuelOsborne) [inline packages/common/src/dotlottie-player.ts:311]: im not sure we should be dividing by 0 if totalFrames is null?

File: packages/common/src/dotlottie-player.ts [hunk 2]
```diff
@@ -198,6 +210,12 @@ export class DotLottieContainer {
     this.setCurrentState(PlayerState.Paused);
   }
 
+  public freeze(): void {
+    if (!this._lottie) return;
+    this._lottie.pause();
+    this.setCurrentState(PlayerState.Frozen);
+  }
+
   public destory(): void {
```
Comment (samuelOsborne) [inline packages/common/src/dotlottie-player.ts:219]: typo

File: packages/common/rollup.config.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+const configs = [
+  bundle({
+    plugins: [
+      del({ targets: 'dist/*' }),
+      resolve({
+        extensions,
+        jsnext: true,
+        module: true,
+      }),
+      webWorkerLoader(),
```
Comment (theashraf) [inline packages/common/rollup.config.js:36]: do we still need the `webWorkerLoader` ?

File: packages/common/rollup.config.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import packageJson from './package.json' assert { type: 'json' };
+
+const extensions = ['.ts', '.tsx'];
+
+const bundle = (config) => ({
+  ...config,
+  input: './src/index.ts',
+  external: (id) =>
+    !/^[./]/u.test(id) &&
+    !/^web-worker.*/u.test(id) &&
```
Comment (theashraf) [inline packages/common/rollup.config.js:22]: do we still need to bundle the `web-worker` ?

File: packages/react-player/.npmrc [hunk 5]
```diff
@@ -0,0 +1,2 @@
+@lottiefiles:registry=https://npm.pkg.github.com
+@dotlottie:registry=https://npm.pkg.github.com
```
Comment (theashraf) [inline packages/react-player/.npmrc:2]: why do we need the `@dotlottie` scope and githhub npm registry ? are we installing any private packages from `@dotlottie` org ?

File: packages/react-player/rollup.config.js [hunk 6]
```diff
@@ -14,10 +18,14 @@ const bundle = (config) => ({
   ...config,
   input: './src/index.ts',
   external: (id) =>
-    !/^[./]/.test(id) && !/^web-worker.*/.test(id) && !/^fflate/.test(id) && !/^@lottiefiles\/dotlottie-js/.test(id),
+    !/^[./]/u.test(id) &&
+    !/^web-worker.*/u.test(id) &&
```
Comment (theashraf) [inline packages/react-player/rollup.config.js:22]: is there a worker in the react player component ?

File: packages/react-player/.npmrc (hunk 5 above)
Comment (afsal-ali) [reply packages/react-player/.npmrc:2]: oh yeah. only using @lottiefiles. Removing it.

**Pull request merged**